tiledb/sm/filter/compression_filter.cc

```diff
@@ -396,6 +435,144 @@ Status CompressionFilter::decompress_part(
+  }
+
+  // special case for the last string
+  input_view[i] = std::string_view(
+      input_buf + offsets_data[i], input.size() - offsets_data[i]);
+
+  return input_view;
+}
+
+Status CompressionFilter::compress_var_string_coords(
+    const FilterBuffer& input,
+    Tile* const offsets_tile,
+    FilterBuffer& output,
+    FilterBuffer& output_metadata) const {
+  if (input.num_buffers() != 1) {
+    return LOG_STATUS(
+        Status_FilterError("Var-sized string input has to be in single "
+                           "buffer format to be compressed with RLE"));
+  }
+
+  // Construct string view of input
+  auto input_view = create_input_view(input, offsets_tile);
+
+  // Estimate and allocate output size
+  /*
+   * TBD: another option instead of traversing the whole input to get the
+   * parameters to be used for estimating the output size + the datasizes
+   * of run length and string length would be to :
+   * - create an overhead function like the rest of compressors and estimate for
+   * the worst case: output_size = input.size() + input.size() *
+   * (sizeof(run_length_datatype) + sizeof(string_length_datatype))
+   * - estimate run length datasize by picking the one that would fit
+   * input->size(), assuming identical single character strings as the worst
+   * case.
+   * - estimate string length by picking the one that would fit input->size(),
+   * assuming a single string as the worst case.
+   * Such an option would be
+   * potentially more wasteful in terms of storage, but more performant.
+   */
+  uint64_t output_size_ub = 0;
+  uint8_t rle_len_bytesize = 0, string_len_bytesize = 0;
+  auto [max_rle_len, max_string_size, num_of_runs, output_strings_size] =
+      RLE::calculate_compression_params(input_view);
+  // TODO: calculate smallest datatypes to fit max_string_size and
+  // max_rle_len -> hardcode for now
+  rle_len_bytesize = 4;
+  string_len_bytesize = 2;
+  output_size_ub = num_of_runs * (rle_len_bytesize + string_len_bytesize) +
+                   output_strings_size;
+
+  // Allocate output data buffer
+  RETURN_NOT_OK(output.prepend_buffer(output_size_ub));
+  Buffer* data_buffer = output.buffer_ptr(0);
+  assert(data_buffer != nullptr);
+  data_buffer->reset_offset();
+  auto output_view = span<std::byte>(
+      reinterpret_cast<std::byte*>(data_buffer->data()), output_size_ub);
+
+  switch (compressor_) {
+    case Compressor::RLE: {
+      // TODO : same as above, hardcode for now
+      RLE::compress<uint32_t, uint16_t>(input_view, output_view);
+      break;
+    }
+    default:
+      break;
+  }
+
+  data_buffer->set_size(output_size_ub);
+
+  // Note: assumes single buffer (holds when RLE is the first/only filter)
+  auto input_size = input.buffers()[0].size();
+  RETURN_NOT_OK(output_metadata.write(&input_size, sizeof(uint32_t)));
+  RETURN_NOT_OK(output_metadata.write(&output_size_ub, sizeof(uint32_t)));
+  // TBD : we could pack those bytesizes in 1 byte instead, do we want that?
```

Comment:
Discussed with @ypatia , good idea but we decided 1 byte savings wasn't huge and readability of the code was worth 1 byte extra per tile.